File: Chapter 03/JS/section_3_3_5_dip/after/midfielder.js
/**
 * A Low-Level Detail.
 * * ARCHITECTURE NOTE: The Midfielder is now a "plug-and-play" component. 
 * It no longer has a unique interface that the Coach must specifically 
 * account for.
 */
class Midfielder {
    performAction() {
        console.log("  [Action] Midfielder is controlling the game.");
    }
}

module.exports = Midfielder;

File: Chapter 03/Python/section_3_3_5_dip/after/midfielder.py
from .player import Player

"""
A Low-Level Detail.
* ARCHITECTURE NOTE: By inheriting from the Player ABC, these classes 
explicitly sign the contract. They are now "plug-and-play" components 
that the Coach can lead without needing to know their specific types.
"""
class Midfielder(Player):
    def perform_action(self):
        print("  [Action] Midfielder is controlling the game.")